node/pkg/common/queryRequest.go

```diff
@@ -14,15 +14,63 @@ import (
 	ethCrypto "github.com/ethereum/go-ethereum/crypto"
 )
 
+// QueryRequest defines a cross chain query request to be submitted to the guardians.
+// It is the payload of the SignedQueryRequest gossip message.
+type QueryRequest struct {
+	Nonce           uint32
+	PerChainQueries []*PerChainQueryRequest
+}
+
+// PerChainQueryRequest represents a query request for a single chain.
+type PerChainQueryRequest struct {
+	// ChainId indicates which chain this query is destine for.
+	ChainId vaa.ChainID
+
+	// Query is the chain specific query data.
+	Query ChainSpecificQuery
+}
+
+// ChainSpecificQuery is the interface that must be implemented by a chain specific query.
+type ChainSpecificQuery interface {
+	Type() ChainSpecificQueryType
+	Marshal() ([]byte, error)
+	Unmarshal(data []byte) error
+	UnmarshalFromReader(reader *bytes.Reader) error
+	Validate() error
+}
+
+// ChainSpecificQueryType is used to interpret the data in a per chain query request.
+type ChainSpecificQueryType uint8
```

Comment:
What kind of values do you think will be used here?  Is there any way it can explode to someone wanting more than 256 different values?

Reply:
It's hard to imagine more than 255 query types? (You know, like it was hard to imagine Unix lasting longer than 2^32 seconds since 1970. . .) Seriously though, I think this is fine.

Reply:
worst case, 255 can always be an "extended" type with more sub-types.